GVFS/GVFS.Notifications/VFSForGit.Mac/StatusMenuItem/Utilities/VFSProcessRunner.h

```diff
@@ -0,0 +1,17 @@
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef NSTask *_Nonnull (^ProcessFactory)(void);
+
+@interface VFSProcessRunner : NSObject
+
+- (instancetype _Nullable)initWithProcessFactory:(ProcessFactory)processFactory;
+- (BOOL)tryRunExecutable:(NSURL *)path
+                    args:(NSArray<NSString *> *_Nullable)args
+                  output:(NSString *_Nullable __autoreleasing *_Nonnull)output
```

Comment:
Is this `_Nonnull` required?  I saw that we have `NS_ASSUME_NONNULL_BEGIN` specified on line 3 (Note that I am completely new to this, and so the `_Nonnull` might be required and I just don't understand why 😃)

Reply:
ya, it is required. `NS_ASSUME_NONNULL_BEGIN` does not add nullability for double pointers, this is documented under "Audited Regions -> exceptions" in [this document](https://developer.apple.com/swift/blog/?id=25) - 
`More complex pointer types like id * must be explicitly annotated. For example, to specify a non-nullable pointer to a nullable object reference, use _Nullable id * _Nonnull.
`